services/searchengine/searchengine.go

```diff
@@ -0,0 +1,73 @@
+package searchengine
+
+import (
+	"github.com/mattermost/mattermost-server/v5/jobs"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/services/searchengine/bleveengine"
+	"github.com/mattermost/mattermost-server/v5/services/searchengine/nullengine"
+)
+
+func NewSearchEngineBroker(cfg *model.Config, jobServer *jobs.JobServer) (*SearchEngineBroker, error) {
+	broker := &SearchEngineBroker{
+		cfg:       cfg,
+		jobServer: jobServer,
+	}
+
+	if *cfg.BleveSettings.EnableIndexing && *cfg.BleveSettings.IndexDir != "" {
+		bleveEngine, err := bleveengine.NewBleveEngine(cfg, jobServer)
```

Comment:
I just wanted to point out a small API design thing, which would make this more idiomatic Go. I know there are already existing instances of this. But it would be good to change it. The issue is stutter (https://golang.org/doc/effective_go.html#package-names).

`bleveengine.NewBleveEngine` stutters. The `BleveEngine` part later is not required. It is qualified by the package name itself. I would suggest to make it `bleveengine.New`

Reply:
Very good point, thanks for the heads up here :)